drivers/sensor/lsm6dsl/Kconfig

```diff
@@ -22,26 +22,82 @@ config LSM6DSL_DEV_NAME
 	depends on LSM6DSL
 	default "LSM6DSL"
 
+choice
+	prompt "Interface to AP"
+	depends on LSM6DSL
+	default LSM6DSL_SPI
+	help
+	  Select interface the LSM6DSL driver is connected to AP
+
+config LSM6DSL_SPI
+	depends on SPI
+	bool "SPI Interface"
+
+config LSM6DSL_I2C
+	depends on I2C
+	bool "I2C Interface"
+
+endchoice
+
 config LSM6DSL_I2C_ADDR
 	hex
 	prompt "LSM6DSL I2C address"
-	depends on LSM6DSL
+	depends on LSM6DSL && LSM6DSL_I2C
 	default 0x6A
 	range 0x6A 0x6B
 	help
 	  I2C address of the LSM6DSL sensor.
 	  Use 0x6A if the SA0 pin is pulled to GND or 0x6B if the SA0 pin
 	  is pulled to VCC.
 
-config LSM6DSL_I2C_MASTER_DEV_NAME
-	string "I2C master where LSM6DSL chip is connected"
+config LSM6DSL_SPI_SELECT_SLAVE
+	hex "LSM6DSL SPI slave select pin"
+	depends on LSM6DSL && LSM6DSL_SPI
+	default 1
+	help
+	  LSM6DSL SPI chip select pin (active low).
+
+config LSM6DSL_SPI_BUS_FREQ
+	int "LSM6DSL SPI bus speed in Hz"
+	default 8000000
+	depends on LSM6DSL && LSM6DSL_SPI
+	help
+	  This is the maximum supported SPI bus frequency. The chip supports a
+	  frequency up to 10MHz.
+
+config LSM6DSL_SPI_GPIO_CS
+	bool "LSM6DSL SPI CS through a GPIO pin"
+	default n
+	depends on LSM6DSL && LSM6DSL_SPI
+	help
+	  This option is useful if one needs to manage SPI CS through a GPIO
+	  pin to by-pass the SPI controller's CS logic.
+
+config LSM6DSL_SPI_GPIO_CS_DRV_NAME
+	string "GPIO driver's name to use to drive SPI CS through"
+	default ""
+	depends on LSM6DSL_SPI_GPIO_CS
+	help
+	  This option is mandatory to set which GPIO controller to use in order
+	  to actually emulate the SPI CS.
+
+config LSM6DSL_SPI_GPIO_CS_PIN
+	int "GPIO PIN to use to drive SPI CS through"
+	default 0
+	depends on LSM6DSL_SPI_GPIO_CS
+	help
+	  This option is mandatory to set which GPIO pin to use in order
+	  to actually emulate the SPI CS.
+
+config LSM6DSL_MASTER_DEV_NAME
```

Comment:
You cannot rename this config as is.
"CONFIG_LSM6DSL_I2C_MASTER_DEV_NAME" is generated from device tree (when device tree option is available)
So you should keep CONFIG_LSM6DSL_I2C_MASTER_DEV_NAME and add  CONFIG_LSM6DSL_SPI_MASTER_DEV_NAME

Reply:
ok, got it. I had now to take a deeper look to all this stuff looking at another board. Still I'm not getting all the points, but some of the puzzle pieces are starting to be attached together.